### DISCUSSION TITLE
build and compiling packages for Termux

### DISCUSSION THREAD

**Discussion by JanuszChmiel** [termux/termux-app#2371] [Show and tell]
Dear C and C++ experienced programmers here. I would like to know, if is it possible to build such complex libraryes like Pygobject directly from Termux? Or if such C libraries would had to be modified on many source files parts to run it? I would like to build such apps such as Goodvibes. I know about Proot and Linux distributions. But some devices have non standard Android kernel functions. So Proot must perform costly work around. The question is, if is it possible to build such apps with no need to recode complex C or C++ source codes. I have tried to build goodvibes INternet radio player. Unfortunately, build script could not locate GTK3+. I would like to know, if my attempt have some sense, or C and C++ deep knowledge is really necessary to prepare some apps which can be executed directly from Termux with no need to use Proot.
If there would be someone who would have some time to build some apps with Me on Termux, we could cooperate on this mission.
I would learn something new thanks to you.

**Comment by agnostic-apollo** [comment]
https://github.com/termux/x11-packages/issues/381
